src/index.js

```diff
@@ -131,20 +130,59 @@ function addTask() {
 
 // remove task
 
-// edit task
+// newly comein
+function saveTaskCompletedState(taskElement, isCompleted) {
+  const taskId = taskElement.getAttribute('id');
+  const taskData = {
+    completed: isCompleted,
+  };
+  localStorage.setItem(taskId, JSON.stringify(taskData));
+}
+
+function markTaskAsCompleted(taskElement) {
+  taskElement.classList.add('completed');
+  const checkbox = taskElement.querySelector('input[type="checkbox"]');
+  checkbox.checked = true;
+  saveTaskCompletedState(taskElement, true);
+}
+
+function markTaskAsUncompleted(taskElement) {
+  taskElement.classList.remove('completed');
+  const checkbox = taskElement.querySelector('input[type="checkbox"]');
+  checkbox.checked = false;
+  saveTaskCompletedState(taskElement, false);
+}
 
 function removeCompletedTasks() {
   const checkboxes = document.querySelectorAll('input[type="checkbox"]');
   checkboxes.forEach((checkbox) => {
+    checkbox.addEventListener('change', function () {
+      if (this.checked) {
+        const li = this.parentNode;
+        if (li.classList.contains('completed')) {
+          markTaskAsUncompleted(li);
+        } else {
+          markTaskAsCompleted(li);
+        }
+      } else {
+        const li = this.parentNode;
+        if (li.classList.contains('completed')) {
+          markTaskAsUncompleted(li);
+        } else {
+          saveTaskCompletedState(li, false);
```

Comment:
Great job @SAZZAD1Q2 🤝

- [ ] [Optional] Kindly ensure the clear completed button works as expected. Whenever I try to clear the selected items nothing seems to happen.